autopilot/contracts.go

```diff
@@ -39,3 +77,244 @@ func (ap *Autopilot) defaultContracts() ([]worker.Contract, error) {
 	}
 	return contracts, nil
 }
+
+func (ap *Autopilot) contractLoop() {
+	for {
+		select {
+		// TODO: use build var for loop interval
+		// TODO: add wakeChan (to trigger loop)
+		// TODO: run loop immediately
+		// TODO: logging
+		case <-ap.stopChan:
+			return
+		case <-time.After(time.Hour):
+		}
+
+		// don't run the contract loop if we are not synced
+		if !ap.store.Synced() {
+			continue
+		}
+
+		// fetch all contracts
+		cs, err := ap.defaultContracts()
+		if err != nil {
+			logErr(err)
+			continue
+		}
+
+		// run checks on the contract set
+		// TODO: dedupe existing contracts (?)
```

Comment:
If we track sectors with their contract IDs we probably don't need to dedupe contracts in the same period. Because it won't mess up our migrations.
We should make sure renews are atomic though so we never end up with an old and a new contract being both active.